### PR TITLE
BREAKING: Update CurrencyId

### LINKED ISSUE #739: Add KAR & KSM to TokenSymbol
Add KAR and make it native token for Karura
Add KSM and make it relaychain token for Karura
Add LKSM and make it liquid relaychain token for Karura

### PR DESCRIPTION
**Pull request by xlc**
TODOs:
- [x] Update runtimes
- [ ] Update acala.js & frontend
- [x] Update predeploy contracts

Closes #739

### CODE REVIEW

**Comment by zjb0807** [comment]
Unit test to generate Tokens.json needs to debug.

**Review by xlc — COMMENTED** [review]
SDN is for Kusama and PLM is for Polkadot (I got it wrong it previous commit)

File: primitives/src/currency.rs [hunk 1]
```diff
@@ -0,0 +1,181 @@
...
+			type Error = ();
+			fn try_from(v: Vec<u8>) -> Result<CurrencyId, ()> {
+				match v.as_slice() {
+					$(bstringify!(vname) => Ok(CurrencyId::Token(TokenSymbol::$vname)),)*
+					_ => Err(()),
+				}
+			}
+		}
+
+		#[test]
```
Comment (xlc) [inline primitives/src/currency.rs:44]: add `#[ignore]`

File: primitives/src/currency.rs [hunk 2]
```diff
@@ -0,0 +1,181 @@
...
+			let tokens = vec![
+				$(
+					Token {
+						name: stringify!($vname).to_string(),
+						symbol: stringify!($vname).to_string(),
+						currencyId: $val.to_string(),
+					},
+				)*
+			];
+			frame_support::assert_ok!(std::fs::write("tokens.json", serde_json::to_string_pretty(&tokens).unwrap()));
```
Comment (xlc) [inline primitives/src/currency.rs:63]: can make path to the predeploy-contracts submodule directly so we don't need to move files. we can just run this and commit the change

Comment (zjb0807) [reply primitives/src/currency.rs]: Add `scripts/generate-tokens-and-predeploy-contracts.sh` to do it.

**Pull request merged**